### PR TITLE
Bounty Hunters have personalized arrival announcements

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Fugitive Hunters now have a personalized message for when they arrive, replacing the generic "Unidentified ship detected near the station" message.

Spacepol:

image

Russians:

image

Bounty Hunters:

image

Psykers:

image

I swear, someday I'll datumize these backstories like we did for pirates. Maybe after this PR is resolved.
## Why It's Good For The Game

Added flavor is nice. It also gives the crew a cue on how to react to the hunters, seeing as some hunters are received far more cooperatively than others (looking at you, SpacePol).

The generic message used to be a fakeout for pirates, but now that THEY have personalized messages, it doesn't really need to be like that anymore.
## Changelog
:cl:
add: Bounty Hunter teams now have personalized announcements for when they are spawned in.
/:cl:

### CODE REVIEW

**Comment by Rhials** [comment]
Probably worth noting that this will collide with #84144. It's what inspired me to make this in the first place. We'll have to see which gets merged first.

File: code/modules/events/ghost_role/fugitive_event.dm [hunk 1]
```diff
@@ -152,6 +152,35 @@
...
+
+	if(!length(announcement_text))
+		announcement_text += "Unidentified ship detected near the station."
+		stack_trace("Fugitive hunter announcement was unable to generate an announcement text based on backstory: [backstory]")
+
+	if(!length(announcement_title))
+		announcement_title += "Unknown Signal"
+		stack_trace("Fugitive hunter announcement was unable to generate an announcement title based on backstory: [backstory]")
+
+	priority_announce(announcement_text, announcement_title)
```
Comment (san7890) [inline code/modules/events/ghost_role/fugitive_event.dm:184]: string tree balancing reeeee just do something like

```dm
var/announcement_title = ""
var/list/announcement_text = list()
switch(backstory)
 if(HUNTER_WHATEVER)
  announcement_title = "Important Thing"
  announcement_text += "I like"
  announcement_text += "big chungus"
  announcement_text += "memes!"
 // rest of code goes here

// length checks fine here as well

priority_announce(jointext(announcement_text_strings, " "), announcement_title)
```

much better and you also don't have to do those dumb leading whitespace things in your strings anyways

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/events/ghost_role/fugitive_event.dm [hunk 2]
```diff
@@ -152,6 +152,35 @@
...
+			announcement_text_list += "Zdraviya zhelaju, [GLOB.station_name] crew. We are coming to your station."
+			announcement_text_list += "There is a criminal aboard. We will arrest them and return them to the gulag. That's good, yes?"
+			announcement_title += "Russian Freighter"
+		if(HUNTER_PACK_BOUNTY)
+			announcement_text_list += "[GLOB.station_name]. One of our bounty marks has ended up on your station. We will be arriving to collect shortly."
+			announcement_text_list += "Let's make this quick. If you don't want trouble, stay the hell out of our way."
+			announcement_title += "Unregistered Signal"
+		if(HUNTER_PACK_PSYKER)
+			announcement_text_list += "HEY, CAN YOU HEAR US? We're coming to your station. There's a bad guy down there, really bad guy. We need to arrest them."
+			announcement_text_list += "We're also offering fortune telling services out from the front door, if you have paying customers."
```
Comment (jlsnow301) [inline code/modules/events/ghost_role/fugitive_event.dm:173]:
```suggestion
			announcement_text_list += "We're also offering fortune telling services out of the front door if you have paying customers."
```

File: code/modules/events/ghost_role/fugitive_event.dm [hunk 3]
```diff
@@ -152,6 +152,35 @@
...
+		if(HUNTER_PACK_RUSSIAN)
+			announcement_text_list += "Zdraviya zhelaju, [GLOB.station_name] crew. We are coming to your station."
+			announcement_text_list += "There is a criminal aboard. We will arrest them and return them to the gulag. That's good, yes?"
+			announcement_title += "Russian Freighter"
+		if(HUNTER_PACK_BOUNTY)
+			announcement_text_list += "[GLOB.station_name]. One of our bounty marks has ended up on your station. We will be arriving to collect shortly."
+			announcement_text_list += "Let's make this quick. If you don't want trouble, stay the hell out of our way."
+			announcement_title += "Unregistered Signal"
+		if(HUNTER_PACK_PSYKER)
+			announcement_text_list += "HEY, CAN YOU HEAR US? We're coming to your station. There's a bad guy down there, really bad guy. We need to arrest them."
```
Comment (jlsnow301) [inline code/modules/events/ghost_role/fugitive_event.dm:172]: i read this like donald trump

**Review by Ghommie — APPROVED** [review]

**Pull request merged**